x-pack/plugins/reporting/public/components/reporting_panel_content.tsx

```diff
@@ -85,7 +90,7 @@ class ReportingPanelContentUi extends Component<Props, State> {
   }
 
   public render() {
-    if (this.isNotSaved() || this.props.isDirty || this.state.isStale) {
+    if (this.mustSaveReport()) {
```

Comment:
I think we can simply add a prop to the `ReportingPanelContent` component called `requiresSavedState` or something. In `x-pack/plugins/reporting/public/share_context_menu/register_csv_reporting.tsx#L109`, the prop is hardcoded to `false` and in `x-pack/plugins/reporting/public/components/screen_capture_panel_content.tsx#L45` it is hardcoded to `true`. 

```suggestion
    if (this.props.requiresSavedState() && this.isNotSaved() || this.props.isDirty || this.state.isStale) {
```